papis/commands/mv.py

```diff
@@ -55,58 +105,92 @@ def run(document: papis.document.Document,
     # Leave this imports here for performance
     import prompt_toolkit
     import prompt_toolkit.completion
 
-    documents = papis.cli.handle_doc_folder_query_sort(query,
-                                                       doc_folder,
-                                                       sort_field,
-                                                       sort_reverse)
+    documents = papis.cli.handle_doc_folder_query_all_sort(query,
+                                                           doc_folder,
+                                                           sort_field,
+                                                           sort_reverse,
+                                                           _all
+                                                           )
     if not documents:
         logger.warning(papis.strings.no_documents_retrieved_message)
         return
 
-    document = documents[0]
-
     lib_dir = os.path.expanduser(papis.config.get_lib_dirs()[0])
-
     completer = prompt_toolkit.completion.PathCompleter(
         only_directories=True,
-        get_paths=lambda: [lib_dir]
+        get_paths=lambda: [str(lib_dir)]
     )
 
-    try:
-        new_folder = os.path.join(
-            lib_dir,
-            prompt_toolkit.prompt(
-                message=(
-                    "Enter directory  : (Tab completion enabled)\n"
-                    "Current directory: ({dir})\n".format(
-                        dir=document.get_main_folder_name()
-                    ) + ">  "
-                ),
+    # If the user hasn't passed ``--folder_name``, we'll prompt for one that
+    # will be used throughout the process.
+    dest_path = None
+    if folder_name is None:
+        try:
+            subfolder = prompt_toolkit.prompt(
+                message="Destination folder: (Tab completion enabled)\n> ",
                 completer=completer,
-                complete_while_typing=True
-            ))
-    except Exception as exc:
-        logger.error("Failed to choose directory.", exc_info=exc)
-        return
-
-    logger.info("New document folder: '%s'.", new_folder)
-
-    if not os.path.exists(new_folder):
-        logger.info("Creating path '%s'.", new_folder)
-        os.makedirs(new_folder, mode=papis.config.getint("dir-umask") or 0o666)
-
-    run(document, new_folder, git=git)
+                complete_while_typing=True)
+            dest_path = os.path.join(lib_dir, subfolder)
+        except Exception as exc:
+            logger.error("Failed to choose directory.", exc_info=exc)
+            return
+
+    if not git and os.path.exists(os.path.join(lib_dir, ".git")):
+        git = papis.tui.utils.confirm("Add the changes to git?")
+
+    moves = []
+    for document in documents:
+        # If the user has passed ``--folder-name``, use that to generate the subfolder
+        if folder_name is not None:
+            subfolder = papis.format.format(folder_name, document)
```

Comment:
This not correct, since the document keys can contain additional path separators. There's similar logic in `papis add` that we should refactor and reuse here.

For example, if the title is `Face/Off` and the folder format is `{doc[year]}/{doc[title]}` we'll get unexpected results.

Reply:
Yes! And also special characters. I'll see what `add` does and adapt for here.

Reply:
I've tried to do some refactoring in #829 to extract that logic. Probably needs a bit more work, but it should be doable to use it here as well.